project-keeper/src/main/java/com/exasol/projectkeeper/config/ProjectKeeperConfig.java

```diff
@@ -33,6 +34,65 @@
         MAVEN
     }
 
+    /**
+     * interface for classes that tell PK how to find out the overall project version.
+     */
+    public interface VersionProvider {
```

Comment:
This name is misleading. From a `VersionProvider` i would expect a `get()` method that returns the version.
Maybe rename to `VersionConfig`?